tern/__main__.py

```diff
@@ -73,8 +75,18 @@ def do_main(args):
         if args.dockerfile:
             run.execute_dockerfile(args)
         if args.docker_image:
-            run.execute_docker_image(args)
-        logger.debug('Report completed.')
+            if common.check_tar(args.docker_image):
+                logger.error("%s", errors.incorrect_raw_option)
+            else:
+                run.execute_docker_image(args)
+                logger.debug('Report completed.')
+        if args.raw_image:
+            if not common.check_tar(args.raw_image):
+                logger.error("%s", errors.invalid_raw_image.format(
+                    image=args.raw_image))
```

Comment:
What do you think about adding an option `-t, --type` and one can enter a type like `docker` or `oci` and then one can provide either `-i, --image` and for this input you can detect whether the input is an existing tar archive or a string ?

Reply:
@nishakm I like the `--type` suggestion to determine if the image type is oci or docker, especially for future support of other image formats. The issue I believe lies in automatically detecting image type. I originally designed this PR to use the existing `--image` option to automatically detect whether the input was a tarfile or a string. The trouble I see with doing it this way is that it's impossible to detect the users intent for the image they provided.

If the user enters a typo in the name or path of their tarfile while running Tern, the `check_tar`  function to see if the tarfile is valid will return false, which would thn indicate that it is a string/docker image type input. This would be incorrect but Tern would continue to try to pull "typo.tar" from dockerhub. 

If we want to automatically detect raw tar format vs docker image format, we can do it but we would have to accept the following error to the terminal when we assume that anything not a tarball is a docker hub image tag string (even in the case of typos).

```
2019-10-02 15:06:38,077 - DEBUG - container - Checking if image "typo.tar" is available on disk...
2019-10-02 15:06:38,098 - DEBUG - container - Attempting to pull image "typo.tar"
2019-10-02 15:06:40,404 - WARNING - container - No such image: "typo.tar"
2019-10-02 15:06:40,404 - CRITICAL - report - Cannot find image typo.tar locally or from remote registry.
```

My decision for implementing an explicit `raw` option was to allow Tern to provide more helpful and intuitive error messages. If the user selects the raw option and it is not a valid tarball, Tern can provide specific feedback that it is not of proper tar format.

Reply:
OK, this makes sense considering "STRING" is a different data type than "FILE"